internal/ingress/controller/controller.go

```diff
@@ -185,18 +186,32 @@ func (n *NGINXController) syncIngress(interface{}) error {
 
 	isFirstSync := n.runningConfig.Equal(&ingress.Configuration{})
 	go func(isFirstSync bool) {
+		steps := 1
 		if isFirstSync {
-			glog.Infof("Initial synchronization of the NGINX configuration.")
-
-			// it takes time for NGINX to start listening on the configured ports
+			// For the initial sync it always takes some time for NGINX to
+			//  start listening on the configured port (default 18080)
+			//  For large configurations it might take a while so we loop
+			//  and back off
+			steps = 10
```

Comment:
With 10 steps (and other retry configuration below), what's the longest time it can potentially take to finally give up and fail?

Reply:
```
try 1
wait for 0 + rand(1, 1+0.1)*1.5 < 1.65
try 2
wait for prev_duration + rand(1, 1+0.1)*1.5 < 1.65 + rand(1, 1+0.1)*1.5 < 1.65 * 2
try 3
wait for < 1.65 * 3
try 4
wait for < 1.65 * 4
== by this time it would take < 1.65 + 1.65 * 2 + 1.65 * 3 ==
...
try 9
wait will be < 1.65 * 9
try last time
exit
```

so by the time it's trying n'th time it'd have waited 1.65 * (1 + 2 + ... + n - 1) seconds worst case. So setting n to 10, it would wait 45 seconds before it fails.

can you confirm it will work like above? Is this realistic? It seems to me settings steps to 10 is a bit too much.

Reply:
@ElvinEfendi here
```package main

import (
	"fmt"
	"time"

	"k8s.io/apimachinery/pkg/util/wait"
)

func main() {
	retry := wait.Backoff{
		Steps:    10,
		Duration: 1 * time.Second,
		Factor:   1.5,
		Jitter:   0.1,
	}

	start := time.Now()
	wait.ExponentialBackoff(retry, func() (bool, error) {
		fmt.Printf("%s\n", time.Now().Sub(start).Round(time.Millisecond))
		return false, nil
	})

	fmt.Printf("Total time: %v\n", time.Now().Sub(start).Round(time.Millisecond))
}
```

```
go run main.go 
0s
1.061s
2.702s
5.102s
8.625s
13.902s
22.018s
33.483s
50.837s
1m16.715s
Total time: 1m16.715s
```

I think five step is more than enough

Reply:
Or use a factor of 1.1
```
go run main.go 
0s
1.061s
2.264s
3.555s
4.944s
6.471s
8.192s
9.975s
11.955s
14.119s
Total time: 14.119s
```

Reply:
thanks @aledbf